crates/trie/parallel/src/proof.rs

```diff
@@ -171,6 +173,7 @@ where
         {
             match account_node {
                 TrieElement::Branch(node) => {
+                    branch_node_hash_masks.insert(node.key.clone(), node.hash_mask);
```

Comment:
you should retain updates rather than doing this, because only branch nodes that are not invalidated by prefix set would be present here